src/AutoMapper/TypeMap.cs

```diff
@@ -328,4 +338,39 @@ private void ApplyInheritedTypeMap(TypeMap inheritedTypeMap)
             }
         }
     }
+
+    public class TypeMap<TSource, TDestination>

```

Comment:
This was taken from looking at Mapster's code originally.  The idea is since this is a concrete type, it's faster to access than calling ResolveTypeMap in configurationProvider.  If you go through configurationProvider, you have to search a dictionary, where as here you have direct access to the object.

This isn't just for exceptions.  This is needed for all the custom expressions for type map and its property maps.  You don't want to have to call ResolveTypeMap and create a new TypePair every time a condition needs to be called when mapping.

For Mapster it makes it faster to find the mapping expression after created.  No need to re-look for type maps if it's already stored somewhere you can access.  Obviously we can't exactly the same effect, since we have to call `GetType()` and then can't easily call a generic method.